packages/story-editor/src/app/media/utils/useMediaInfo.js

```diff
@@ -0,0 +1,201 @@
+          const reader = new FileReader();
+          reader.onerror = reject;
+          reader.onload = (event) => {
+            if (event.target.error) {
+              reject(event.target.error);
+            }
+            resolve(new Uint8Array(event.target.result));
+          };
+          reader.readAsArrayBuffer(file.slice(offset, offset + chunkSize));
+        });
+
+      //eslint-disable-next-line @wordpress/no-unused-vars-before-return -- False positive because of the finally().
+      const trackTiming = getTimeTracker('load_mediainfo');
+
+      try {
+        // This will expose the window.MediaInfo global.
+        await loadScriptOnce(mediainfoUrl);
+
+        const mediaInfo = await window.MediaInfo({ format: 'JSON' });
+        const result = JSON.parse(
+          await mediaInfo.analyzeData(getSize, readChunk)
+        );
+
+        const normalizedResult = result.media.track.reduce(
+          (acc, track) => {
+            if (track['@type'] === 'General') {
+              acc.fileSize = Number(track.FileSize);
+              acc.format = track.Format.toLowerCase().replace('mpeg-4', 'mp4');
+              acc.frameRate = Number(Number(track.FrameRate).toFixed(0));
+              acc.codec = track.CodecID?.trim();
+            }
+
+            if (track['@type'] === 'Image' || track['@type'] === 'Video') {
+              acc.width = Number(track.Width);
+              acc.height = Number(track.Height);
+              acc.colorSpace = track.ColorSpace; // Maybe useful in the future.
+            }
+
+            if (track['@type'] === 'Video') {
+              acc.duration = Number(track.Duration);
+              acc.videoCodec = track.Format.toLowerCase();
+            }
+
+            if (track['@type'] === 'Audio') {
+              acc.audioCodec = track.Format.toLowerCase();
+            }
+
+            return acc;
+          },
+          {
+            mimeType: file.type,
+          }
+        );
+
+        normalizedResult.isMuted = !normalizedResult.audioCodec;
+
+        mediaInfo.close();
+
+        return normalizedResult;
+      } catch (err) {
+        // eslint-disable-next-line no-console -- We want to surface this error.
+        console.error(err);
+
+        trackError('mediainfo', err.message);
+
+        return null;
+      } finally {
+        trackTiming();
+      }
+    },
+    [mediainfoUrl]
+  );
+
+  const isConsideredOptimized = useCallback((fileInfo) => {
+    if (!fileInfo) {
+      return false;
+    }
+
+    const hasSmallFileSize =
+      fileInfo.fileSize < MEDIA_VIDEO_FILE_SIZE_THRESHOLD;
+    const hasSmallDimensions =
+      fileInfo.width * fileInfo.height <=
+      MEDIA_VIDEO_DIMENSIONS_THRESHOLD.WIDTH *
+        MEDIA_VIDEO_DIMENSIONS_THRESHOLD.HEIGHT;
+
+    // Video is small enough and has an allowed mime type, upload straight away.
+    const result =
+      hasSmallFileSize &&
+      hasSmallDimensions &&
+      ['video/webm', 'video/mp4'].includes(fileInfo.mimeType);
+
```

Comment:
So we don't check format, frameRate, audioCodec or file size here? Is this something for the future? Maybe add some todos here?

Reply:
Let me reply at https://github.com/GoogleForCreators/web-stories-wp/pull/11454/files#r912080543